src/planner.rs

```diff
@@ -18,12 +15,9 @@ impl RunArg {
     pub fn fetch<'a, U, F>(&'a self, f: F) -> U
         where F: FnOnce(&'a World) -> U
     {
-        let pulse = self.pulse.borrow_mut().take()
-                        .expect("fetch may only be called once.");
-        let u = f(&self.world);
-        pulse.pulse();
-        u
+        f(&self.world)
```

Comment:
@kvark Yes you are right, I didn't really understand how it works.

> The fetch phase execution is serialized

Do you mean sequentialized?